zipkin-storage/elasticsearch/src/test/java/zipkin2/elasticsearch/integration/ElasticsearchStorageExtension.java

```diff
@@ -85,25 +83,30 @@
       return;
     }
 
-   if (container != null) {
-     LOGGER.info("Stopping docker image " + image);
-     container.stop();
-   }
+    if (container != null) {
+      LOGGER.info("Stopping docker image " + image);
+      container.stop();
+    }
   }
 
-  void tryToInitializeSession() {
+  void tryToInitializeSession() throws IOException {
     try (ElasticsearchStorage result = computeStorageBuilder().build()) {
       CheckResult check = result.check();
       assumeTrue(check.ok(), () -> "Could not connect to storage, skipping test: "
         + check.error().getMessage());
     }
   }
 
-  ElasticsearchStorage.Builder computeStorageBuilder() {
-    ElasticsearchStorage.Builder builder = ElasticsearchStorage.newBuilder()
+  Builder computeStorageBuilder() {
+    Builder builder = ElasticsearchStorage.newBuilder(() ->
+      // Elasticsearch 7 never returns a response when receiving an HTTP/2 preface instead of the
```

Comment:
this is the only special case needed for functionality to work. this means the only hard opinion about client setup we have is about an ES bug not yet known or raised.